extensions/gdata/tests/src/com/google/refine/extension/gdata/GoogleAPIExtensionTest.java

```diff
@@ -0,0 +1,38 @@
+package com.google.refine.extension.gdata;
+
+import com.google.refine.ProjectManager;
+import org.testng.annotations.Test;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.*;
+
+public class GoogleAPIExtensionTest {
+
+	@Test
+	public void reTest() {
+		ProjectManager.singleton = mock(ProjectManager.class);
```

Comment:
This assignment of `ProjectManager.singleton` does not seem to be needed. Perhaps the test could also have a more explanatory name (making it clearer that it relates to `extractSpreadSheetId`).

Reply:
We do need `ProjectManager.singleton = mock(ProjectManager.class)`, because GoogleAPIExtension.&lt;clinit&gt; will call `ProjectManager.singleton.getPreferenceStore()`, which will result in NullPointerException if `singleton` is null.

I'll write a comment for this in the test.